src/operators/select/select.cc

```diff
@@ -126,159 +127,105 @@ arrow::Datum Select::Filter(const std::shared_ptr<Block> &block,
 
 arrow::Datum Select::Filter(const std::shared_ptr<Block> &block,
                             const std::shared_ptr<Predicate> &predicate) {
-  switch (predicate->value_.type()->id()) {
-    case arrow::Type::UINT8: {
-      uint8_t val = std::static_pointer_cast<arrow::UInt8Scalar>(
-                        predicate->value_.scalar())
-                        ->value;
-      auto datum_val = predicate->value_;
-
-      switch (predicate->comparator_) {
-        case arrow::compute::CompareOperator::LESS: {
-          return Filter<uint8_t>(block, predicate->col_ref_, datum_val, val,
-                                 predicate->comparator_, std::less());
-        }
-        case arrow::compute::CompareOperator::LESS_EQUAL: {
-          return Filter<uint8_t>(block, predicate->col_ref_, datum_val, val,
-                                 predicate->comparator_, std::less_equal());
-        }
-        case arrow::compute::CompareOperator::GREATER: {
-          return Filter<uint8_t>(block, predicate->col_ref_, datum_val, val,
-                                 predicate->comparator_, std::greater());
-        }
-        case arrow::compute::CompareOperator::GREATER_EQUAL: {
-          return Filter<uint8_t>(block, predicate->col_ref_, datum_val, val,
-                                 predicate->comparator_, std::greater_equal());
-        }
-        case arrow::compute::CompareOperator::EQUAL: {
-          return Filter<uint8_t>(block, predicate->col_ref_, datum_val, val,
-                                 predicate->comparator_, std::equal_to());
-        }
+  auto data_type = predicate->value_.type();
+
+  // [1] Handle default types.
+  // TODO: (Type Coverage) Any concerns over the default types?
+  auto default_filter = [&]<typename T>(T *) -> arrow::Datum {
+    arrow::Status status;
+    arrow::compute::CompareOptions compare_options(predicate->comparator_);
+    arrow::Datum block_filter;
+
+    auto select_col = block->get_column_by_name(predicate->col_ref_.col_name);
+    auto value = predicate->value_;
+
+    status = arrow::compute::Compare(select_col, value, compare_options)
+                 .Value(&block_filter);
+    evaluate_status(status, __FUNCTION__, __LINE__);
+
+    filters_[block->get_id()] = block_filter.make_array();
+    return block_filter;
+  };
+
+  // [2] Handle numeric types.
+  // TODO: There are a lot of assumptions over the type T.
+  auto numeric_filter = [&]<typename T>(T *) -> arrow::Datum {
+    using CType = ArrowGetCType<T>;
+    using ScalarType = ArrowGetScalarType<T>;
+
+    CType val =
+        std::static_pointer_cast<ScalarType>(predicate->value_.scalar())->value;
+    auto datum_val = predicate->value_;
+
+    switch (predicate->comparator_) {
```

Comment:
I feel this switch can be removed with help of a map (comp->primitive comp) or any other means like type switch, maybe we can handle in another PR

Reply:
Done with a comparator switcherr.